src/common/cockpitwebserver.c

```diff
@@ -1371,6 +1381,23 @@ cockpit_web_request_get_protocol (CockpitWebRequest *self)
 gchar *
 cockpit_web_request_get_remote_address (CockpitWebRequest *self)
 {
+  if (self->web_server && self->web_server->forwarded_for_header)
+    {
+      const gchar *forwarded_header = g_hash_table_lookup (self->headers, self->web_server->forwarded_for_header);
+      if (forwarded_header)
+        {
+          /* This isn't really standardised, but in practice, it's a
+           * space separated list and the last item is from the
+           * immediately upstream server.
+           */
+          const gchar *last_space = strrchr (forwarded_header, ' ');
+          if (last_space)
+            return g_strdup (last_space + 1);
```

Comment:
Is there any chance that trailing spaces could happen? (I hope not..) But not sure if this should prevent returning an empty string.

Reply:
I ask myself what a trailing space would mean in that case: I feel like from a practical standpoint that would most likely be the result of a proxy server attempting, and failing, to insert the IP address.

In any case, HTTP is clear about this: if there's trailing whitespace on a header line then it doesn't count as part of the value, and indeed, in our parser code:

```
            g_autofree gchar *value = g_strndup (colon + 1, line - (colon + 1));
            g_strstrip (value);
```

So it's actually impossible for there to be trailing whitespace there.  We can add a test for it, of course.